pkg/crypto/certificatemanager/certificate_manager.go

```diff
@@ -9,37 +9,65 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/cilium/cilium/pkg/hive/cell"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
 	"github.com/cilium/cilium/pkg/policy/api"
+
+	"github.com/spf13/pflag"
 )
 
-const (
-	caDefaultName      = "ca.crt"
-	publicDefaultName  = "tls.crt"
-	privateDefaultName = "tls.key"
+var Cell = cell.Module(
+	"certificate-manager",
+	"Provides TLS certificates and secrets",
+
+	cell.Provide(NewManager),
+
+	cell.Config(defaultManagerConfig),
 )
 
-type k8sClient interface {
-	GetSecrets(ctx context.Context, ns, name string) (map[string][]byte, error)
+type CertificateManager interface {
+	GetTLSContext(ctx context.Context, tlsCtx *api.TLSContext, ns string) (ca, public, private string, err error)
+}
+
+type SecretManager interface {
+	GetSecrets(ctx context.Context, secret *api.Secret, ns string) (string, map[string][]byte, error)
+	GetSecretString(ctx context.Context, secret *api.Secret, ns string) (string, error)
```

Comment:
not really a review point, but GetSecretsString doesn't actually need any internal access to the *CertManager, it could just be a standalone helper func.